repoze/debug/responselogger.py

```diff
@@ -99,7 +99,8 @@ def get_request_info(self, environ):
                 readargs.append(int(environ['CONTENT_LENGTH']))
             info['body'] = environ['wsgi.input'].read(*readargs)
             try:
-                environ['wsgi.input'].seek(0)
+                if hasattr(environ['wsgi.input'], 'seek'):
+                    environ['wsgi.input'].seek(0)
             except AttributeError:
```

Comment:
I don't believe this change can be correct (or at least complete):  the rest of the WSGI stack relies on having the input stream "reset" to its original point (after the copy made above in line 100).  Suppressing the `AttributeError` here (which is immediately caught in line 103) means that the `BytesIO` copy of the stream won't be made, and therefore that the application wrapped by the middleware won't have access to the contents of the stream.